python/whylogs/core/validators/condition_validator.py

```diff
@@ -18,15 +18,15 @@ class ConditionValidator(Validator):
     A validator that checks if a column satisfies a condition.
 
     Args:
-        name (str): The name of the column to validate
+        name (str): The name of the ConditionValidator.
         conditions (Dict[str, Union[Condition, Callable[[Any], bool]]]): A dictionary of conditions to check.
         actions (List[Callable[[str, str, Any, Any], None]]): A list of actions to take when a condition fails.
         enable_sampling (bool): Whether to enable sampling of failed values. Defaults to True.
 
     """
 
     conditions: Dict[str, Union[Condition, Callable[[Any], bool]]]
-    actions: List[Callable[[str, str, Any, Any], None]]
+    actions: List[Union[Callable[[str, str, Any], None], Callable[[str, str, Any, Optional[Any]], None]]]
```

Comment:
I take it the `Union` is for back compatability?

We need to do some cleanup in the `track()` calls in `DatasetProfile`

Reply:
Yes, for back compat - before the identity values addition, action would take only 3 arguments. With identity values, it's now 4.